### PR TITLE
Pda 180

### PR DESCRIPTION
**Pull request by gReis89**

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **ffleandro**

If they are not already assigned, you can assign the PR to them by writing `/assign @ffleandro` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fentando-process-driven-plugin).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/entando-process-driven-plugin/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is well

File: widgets/src/api/makeRequest.js
Comment (rimselis) [inline widgets/src/api/makeRequest.js]:
```suggestion
  if (responseHeaders && responseHeaders.includes('xml')) {
    return response.text();
  }
  if (responseHeaders && responseHeaders.includes('octet')) {
    return response;
  }
  return response.json();
```

File: widgets/src/components/Attachments/AttachmentsContainer.js
Comment (rimselis) [inline widgets/src/components/Attachments/AttachmentsContainer.js:172]: Can't we use the async iterators? E.g.
```
async function getResponseSize(url) {
  const response = await fetch(url);
  let total = 0;

  for await (const chunk of response.body) {
    total += chunk.length;
  }
  return total;
}
```

File: widgets/src/api/makeRequest.js
Comment (gReis89) [reply widgets/src/api/makeRequest.js]: changed the logic to be more readable, thanks!

**Comment by entando-service** [comment]
All is well

**Review by rimselis — APPROVED** [review]

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/entando-process-driven-plugin/pull/99#pullrequestreview-390745697" title="Approved">rimselis</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **ffleandro**

If they are not already assigned, you can assign the PR to them by writing `/assign @ffleandro` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fentando-process-driven-plugin).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/entando-process-driven-plugin/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: widgets/src/components/Attachments/AttachmentsContainer.js
Comment (gReis89) [reply widgets/src/components/Attachments/AttachmentsContainer.js:172]: Talking with Martynas, he suggested a solution from this article:
https://jakearchibald.com/2017/async-iterators-and-generators/#making-streams-iterate

but as it doesn't solve the main issue that is using the `while (true)`, we agreed to keep the implementation for now, and revisit it later.

**Pull request merged**
